fdbserver/Status.actor.cpp

```diff
@@ -2184,6 +2185,12 @@ ACTOR static Future<JsonBuilderObject> workloadStatusFetcher(
 		transactions["committed"] = txnCommitOutSuccess.getStatus();
 
 		statusObj["transactions"] = transactions;
+
+		uint64_t numTenants = 0;
+		if (commitProxyStats.size() > 0) {
```

Comment:
The usual policy for status when a particular piece of data cannot be determined is to omit it from the document rather than give it a default value. Arguably we could give it a default value, but in that case we would want it to be distinct from a valid value.

Reply:
There are a couple of ways of fixing this, as I understand.
1. The simplest fix is to just move line 2193 (ie. (*tenants)["num_tenants"] = numTenants; to be inside the if block.
2. Going further, if the numTenants information cannot be determined from the first commit proxy's stats, try looking at other proxy stats in the list. This leads to the question: Do the proxy stats ever disagree among different proxies? And should we be concerned about that here?

I'll probably leave the refactoring of moving the proxy-stats-fetch to a higher level for later, if that is ok with you.

Reply:
> The simplest fix is to just move line 2193 (ie. (*tenants)["num_tenants"] = numTenants; to be inside the if block.

That sounds right to me

> Going further, if the numTenants information cannot be determined from the first commit proxy's stats, try looking at other proxy stats in the list. This leads to the question: Do the proxy stats ever disagree among different proxies? And should we be concerned about that here?

I think we can probably ignore this issue. It shouldn't be possible for the commit proxy metrics to be logged without a `NumTenants` field, so as long as the event is present we should have some data. It is possible for there to be disagreement between the commit proxies, but it should be transient and any value reported by a commit proxy should be acceptable. It might be slightly stale, but that is true of all of our metrics based on 5s reporting intervals.

> I'll probably leave the refactoring of moving the proxy-stats-fetch to a higher level for later, if that is ok with you.

That's fine. The main consequence of this is that certain sections in the document get tied into the same error scope, but I don't expect that to be a showstopper.